documentation/gitbook/development/python-guidelines-for-developers.md

```diff
@@ -40,16 +40,21 @@ Before committing, make sure you **format your code** using black (see below) an
 
 ### Formatting
 
-We try to remain close to python conventions and to follow the updated rules and best practices. For formatting, we use [black](https://github.com/psf/black), the _Uncompromising Code Formatter_ before submitting pull-requests. It provides a consistent style, which is useful when diffing. We use a custom length of 120 characters as the baseline is short for scientific code. Here is the line to use to format your code:
+{% hint style="info" %}
+Code formatters are necessary, but not sufficient for well formatted code and further style changes may be requested in PRs. Furthermore, the formatting/linting requirements for code contributed to `flepiMoP` are likely to be enhanced in the future and those changes will be reflected here when they come.
+{% endhint %}
+
+For python code formatting the [black](https://black.readthedocs.io/en/stable/) code formatter is applied to all edits to python files being merged into `flepiMoP`. For installation and detailed usage guides please refer to the black documentation. For most use cases the following commands are sufficient:
 
 ```bash
-black --line-length 120 . --exclude renv*
+# See what style changes need to be made
+black --diff .
+# Reformat the python files automatically
+black .
+# Check if current work would be allowed to merged into flepiMoP
+black --check .
```

Comment:
I changed this to use the default line length (88 I think) and I assumed it wouldn't be a problem as I didn't think code formatting was actively being used, but I was wrong: https://github.com/HopkinsIDD/flepiMoP/pull/296#discussion_r1715729293. Do we have strong opinions about keeping this at 120? That's a ~33% increase over the default, which seems too wide to me. Definitely would make diffing two files painful or even viewing a file on the go a bit annoying. However, if there is strong consensus for the 120 I can change this back both here and in the CI action.

CC: @fang19911030, @jcblemai, @pearsonca

Reply:
classic line limits derive from punchcard languages, but modern screen dimensions / human reading comprehension tend to favor shorter line length limits. I recommend 72 or 80, which are the common cross-language defaults, inherited from [fortran](https://stackoverflow.com/questions/19311766/gfortran-line-length-limit) - not aware of python stretching those, but could be?

Reply:
I don't have a preference for these two numbers. I am happy with both of them.

Reply:
4 spaced tabs are used for indentation, so for a method inside a class with say a nested if (or for and then if) that's already 16 characters used up, I think something like 72-80 would leave too little space after that? But I'd be curious to hear other's thoughts. Pylint's default is 100, which is too long as well.

Looping in @emprzy, @shauntruelove, and @saraloo if y'all have any thoughts.

Reply:
I don't have any strong opinions. I agree that 72 is pretty short, but I don't really have too much of a dog in this fight.

Reply:
I really like longer lines, especially as we have a large space indentation. Otherwise, it makes code in things like step_rk4 very hard to read and diff. 120 has been chosen for this.